src/NHibernate/Impl/ISupportEntityJoinCriteria.cs

```diff
@@ -0,0 +1,11 @@
+using NHibernate.Criterion;
+using NHibernate.SqlCommand;
+
+namespace NHibernate.Impl
+{
+	public interface ISupportEntityJoinCriteria
```

Comment:
I would add a " 6.0 TODO: merge with ICriteria". Is there a reason for not aiming at this?